apps/web/src/components/collectibles/collectible-browser.tsx

```diff
@@ -70,15 +71,51 @@ export default function CollectibleBrowser({
   return (
     <div className={css.root}>
       <Heading className={css.title}>{collectible.title}</Heading>
+      {collectible.previewVideo && (
+        <button
+          onClick={() => setShowVideoCoverImage(!showVideoCoverImage)}
+          className={css.flipButton}
+        >
+          {showVideoCoverImage ? 'show video' : 'show cover'}
+        </button>
+      )}
       <div className={css.imageWrapper}>
-        <Image
-          src={collectible.image}
-          loader={cmsImageLoader}
-          width={700}
-          height={700}
-          layout="responsive"
-          objectFit="contain"
-        />
+        {!collectible.previewVideo && (
+          <Image
+            src={collectible.image}
+            loader={cmsImageLoader}
+            width={700}
+            height={700}
+            layout="responsive"
+            objectFit="contain"
+          />
+        )}
+        {collectible.previewVideo && (
+          <div
+            className={clsx(css.flipBox, {
+              [css.flip]: showVideoCoverImage,
+            })}
+          >
+            <div className={css.flipBoxInner}>
+              <div className={css.flipBoxFront}>
+                <video width="100%" controls muted autoPlay loop>
+                  <source src={collectible.previewVideo} type="video/mp4" />
+                  Your browser does not support the video tag.
```

Comment:
Probably worth translating this as well. There's some documentation in the front-end readme as well as some other examples in this file. Essentially, you'll use the `t()` command and pipe it to a translation file (`common:statuses` is probably a good one for this scenario). Happy to show you around the i18n stuff if you've never worked with it before!